source/connect.txt

```diff
@@ -122,6 +122,52 @@ option for programmatic usage of ``mongosh``, like a :driver:`driver
    - To provision access to a MongoDB deployment, see :manual:`Database
      Users </core/security-users/>`.
 
+
+Connect with LDAP
+~~~~~~~~~~~~~~~~~
+
+When using LDAP for authorization, users connecting via
+``mongosh`` must:
+
+- Set :option:`--authenticationDatabase <mongosh --authenticationDatabase>` to ``$external``.
+
+- Set :option:`--authenticationMechanism <mongosh --authenticationMechanism>` to the appropriate authentication
+  mechanism.
+
+  If using :ref:`LDAP authentication <security-ldap>`, set this to ``PLAIN``.
+
+  If using :ref:`Kerberos authentication <security-kerberos>`, set this to
+  ``GSSAPI``.
+
+  If using :ref:`x.509 <security-auth-x509>`, set this to ``MONGODB-X.509``.
+
+- Set :option:`--username <mongosh --username>` to a username that respects the
+  :setting:`security.ldap.authz.queryTemplate`, or any configured
+  :setting:`security.ldap.userToDNMapping` template.
+
+- Set :option:`--password <mongosh --password>` to the appropriate password.
```

Comment:
One problem that occurs with LDAP authentication in mongosh (and to a larger degree also Compass) is that one-time passwords don’t always work cleanly. For example, see the conversation in [HELP-42983](https://jira.mongodb.org/browse/HELP-42983) (or the other tickets linked to [PRODTRIAGE-2022](https://jira.mongodb.org/browse/PRODTRIAGE-2022)).

It might be good to include language here along the lines of “When using one-time passwords with LDAP authentication, consider adding `maxPoolSize=1&srvMaxHosts=1` to your connection string in order to reduce the potential for connection failures.”

(I get that that might sound a bit discouraging to users, having to manually apply a workaround like that; this has come up often enough that I think it might be worth noting it, though, and unfortunately we don’t quite know yet what our long-term story regarding this problem will be.)

Reply:
Thanks for the suggestion, unfortunately facing some build error technical issues so the staging site will not reflect the update. Does this sound ok? It will link [here](https://www.mongodb.com/docs/manual/reference/connection-string/#connection-string-options)

```
   When you use one-time passwords with LDAP authentication, adding 
   the :ref:`connection string options <connections-connection-options>` 
   ``maxPoolSize=1&srvMaxHosts=1`` to your connection string is 
   recommended to reduce the potential for connection failures.
   ```